byterun/caml/roots.h

```diff
@@ -35,10 +35,48 @@ CAMLextern void caml_do_local_roots (scanning_action, value *, value *,
 CAMLextern void caml_do_local_roots(scanning_action f, char * bottom_of_stack,
                                     uintnat last_retaddr, value * gc_regs,
                                     struct caml__roots_block * local_roots);
-#endif
+#endif /* NATIVE_CODE */
 
 CAMLextern void (*caml_scan_roots_hook) (scanning_action);
 
+
+/* Hooks to profile systhreads. */
+
+#define ST_NEXT_HOOK(thread_id,bottom,retaddr) \
```

Comment:
I`m not sure about the choice of names. `st_next_hook` and `st_change_hook`.
What about `st_new` and `st_state_update_hook` ?
Also, I think that both macros should have a short description, not only the enumeration of hooks.